src/App.js

```diff
@@ -1,85 +1,208 @@
+        todoList: [
+          ...prevState.todoList,
+          createdTodo,
+        ],
+      }));
+    }
+  };
+
+  handleCompleted = (event) => {
+    const { todoList } = this.state;
+    const isComplete = event.target.checked;
+    const todoId = event.target.id;
+
+    const changedItem = todoList
+      .find(todo => todo.id === todoId);
+
+    changedItem.completed = isComplete;
+
+    this.setState(prevState => ({
+      todoList: [
+        ...prevState.todoList
+          .slice(0, todoList.indexOf(changedItem)),
+        changedItem,
+        ...prevState.todoList
+          .slice(todoList.indexOf(changedItem) + 1),
+      ],
+    }));
+  };
+
+  handleDestroy = (event) => {
+    const { todoList } = this.state;
+    const { name } = event.target;
+
+    const deletingItem = todoList
+      .find(todo => todo.id === name);
+
+    this.setState(prevState => ({
+      todoList: [
+        ...prevState.todoList
+          .slice(0, todoList.indexOf(deletingItem)),
+        ...prevState.todoList
+          .slice(todoList.indexOf(deletingItem) + 1),
+      ],
+    }));
+  };
+
+  setEditedValue = (value, idx) => {
+    const { todoList } = this.state;
+
+    const changedItem = todoList
+      .find(todo => todo.id === idx);
+
+    if (value.trim()) {
+      changedItem.label = value;
+      changedItem.completed = false;
+
+      this.setState(prevState => ({
+        todoList: [
+          ...prevState.todoList
+            .slice(0, todoList.indexOf(changedItem)),
+          changedItem,
+          ...prevState.todoList
+            .slice(todoList.indexOf(changedItem) + 1),
+        ],
+      }));
+    }
+  };
+
+  handleSelectAll = (isSelectAll) => {
+    const allTodosCompleted = list => (list
+      .map(todo => ({
+        ...todo,
+        completed: isSelectAll,
+      })));
+
+    this.setState(prevState => ({
+      todoList: allTodosCompleted(prevState.todoList),
+    }));
+  };
+
+  clearCompleted = () => {
+    const { todoList } = this.state;
+
+    const newTodos = todoList.filter(todo => !todo.completed);
+
+    this.setState({
+      todoList: newTodos,
+    });
+  };
+
+  handleSort = (name) => {
+    this.setState((prevState) => {
+      if (name === 'active') {
+        return {
+          isSorted: true,
+          sortedTodos: prevState.todoList
+            .filter(todo => !todo.completed),
+        };
+      }
+
+      if (name === 'completed') {
+        return {
+          isSorted: true,
+          sortedTodos: prevState.todoList
+            .filter(todo => todo.completed),
+        };
+      }
+
+      return {
+        isSorted: false,
+        todoList: prevState.todoList,
+        sortedTodos: [],
+      };
+    });
+  };
+
+  render() {
+    const { todoList, isSorted, sortedTodos } = this.state;
+
+    return (
+      <section className="todoapp">
+        <Header
+          setNewTodo={todo => this.setNewTodo(todo)}
+          selectAll={checked => this.handleSelectAll(checked)}
```

Comment:
those are just props, how can they pass any arguments?
just pass the name of the function